### ISSUE TITLE
Cannot set download path

### ISSUE THREAD

**Issue by Serraniel** [flexbooru/flexbooru#110]
**Describe the bug**
When I enter the settings of an image board (doesn´t matter which one) and navigate to its settings I cannot set a download path in the *Advanced / Storage* area

**To Reproduce**
1. Go to any configured image board.
2. Open its settings.
3. Expand the *Advanced* section
4. Click on *Downloadpath* in the *storage* section

**Expected behavior**
I can configure a download path.

**Actual behavior**
Nothing happens

**Screenshots**
[video.zip](https://github.com/flexbooru/flexbooru/files/4215093/video.zip)

**Device (please complete the following information):**
 - Device name: Huawei Mate 20 Pro
 - OS version: Android 10 (Build 10.0.0.180(C432E7R1P5))
 - App version 1.1.10.c874

**Additional context**
App never has asked for storage permissions (which I would have expected). Also have tried to reinstall the app via playstore.

**Comment by onlymash** [comment]
Your ROM missing [com.android.documentsui]

**Comment by onlymash** [comment]
Try if this app can open documentsui
https://play.google.com/store/apps/details?id=org.aospstudio.files

**Comment by Serraniel** [comment]
This App gives an error, too: `The page to which this application is directed is not supported by your manufacturer brand.`.

However, other Apps like Solid File Explorer (which I use normally) have no issues.

**Comment by onlymash** [comment]
flexbooru save file by [SAF](https://developer.android.com/guide/topics/providers/document-provider), No storage permissions required, but SAF relies on documentsui.

**Comment by onlymash** [comment]
Throw away your Huawei and buy another brand

**Comment by Serraniel** [comment]
According to the playstore description of the app you wanted me to install it seems to be a common issue, that this is missing:
> This app may not work on all Android devices.

There also are mentioned many other devices / manufacturers in the reviews which seem to have problems with the app, so I do not think the issue is related to Huawei only. So I kind of think your last comment is a bit inappropriate tbh.

Regarding the issue: do you have any plans to fix this?

**Comment by onlymash** [comment]
This is not my problem, please send feedback to Huawei

**Comment by Serraniel** [comment]
So, I contacted the Huawei support via chat. They will internally forward the issue to another department and want to send a reply in the next days to me, after investigating with the details I provided to them. I will update the issue as far as I got a reply.

However, if the missing module only is required for configuration and not needed for actually saving the images, you maybe could offer some alternatives? Things that came to my mind:
* config file
* default download paths in scheme of `flexbooru/<booru>` 
* some kind of text input (which might be the worst alternative of those three, as there is more potential to bullshit input)
As I do not have the knowledge I cannot evaluate if this will work because I do not know if the download itself will be affected of the issue, too. 

Really would appreciate if you can check the above suggestion. Thanks for your help so far.

**Comment by onlymash** [comment]
Android defines the following permissions for read-and-write access to external storage: READ_EXTERNAL_STORAGE and WRITE_EXTERNAL_STORAGE.(Will no longer be available starting with Android R)

On earlier versions of Android, apps needed to declare these permissions to access any file outside the app-specific directories on external storage. More recent versions of Android rely more on a file's purpose than its location for determining an app's ability to access that file. This purpose-based storage model improves user privacy because apps are given access only to the areas of the device's file system that they actually use.

Via: https://developer.android.com/training/data-storage

So SAF will be the best solution, I will not provide any support for brands that do not follow the rules

**Comment by Serraniel** [comment]
I Had a Phone call with a Dev (or at least Somebody with good and enough technical knowledge) from Huawei and we have been able to understand that the Module Broke during an Android Update (9->10) and possible reasons for that.

Tldr: factory reset has fixed the issue. But still weird Other apps weren't affected at all.
